nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/PutHDFS.java

```diff
@@ -266,6 +268,13 @@ public Object run() {
                             throw new IOException(configuredRootDirPath.toString() + " could not be created");
                         }
                         changeOwner(context, hdfs, configuredRootDirPath, flowFile);
+                    } catch (IOException e) {
+                        if (!Strings.isNullOrEmpty(e.getMessage()) && e.getMessage().contains(String.format("Couldn't setup connection for %s", ugi.getUserName()))) {
```

Comment:
@ekovacs I think we should be more selective in this check.  I don't think there's a better way to detect this error scenario than string matching at this point, but the exception stack should be inspected to see if you can find the GSSException as the root cause:
`Caused by: org.ietf.jgss.GSSException: No valid credentials provided (Mechanism level: Failed to find any Kerberos tgt) `
If you iterate through the causes when PutHDFS encounters an IOException, and see that GSSException, we can do a penalize with a session rollback.
Otherwise, we'd want to pass the flowfile to the failure relationship.

Reply:
Thanks @jtstorck you are right.
I pushed a commit to handle the way you suggested